atoma-proxy-service/src/handlers/auth.rs

```diff
@@ -268,7 +269,10 @@ pub(crate) async fn register(
         .await
         .map_err(|e| {
             error!("Failed to register user: {:?}", e);
-            StatusCode::INTERNAL_SERVER_ERROR
+            match e {
+                AuthError::UserAlreadyRegistered => StatusCode::CONFLICT,
+                _ => StatusCode::INTERNAL_SERVER_ERROR,
```

Comment:
It would be great to have a proper error messaging, maybe for another PR

Reply:
yeah, it would be nice if we return the proper statuscodes, and not internal server error most of the times
IMHO it should be another PR